### PR TITLE
feat(internal-plugin-encryption): support for uploading master key  f…

### PR DESCRIPTION
**Pull request by arun3528**
…or BYOK

# COMPLETES #[SPARK-444636](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-444636)

## This pull request addresses

This PR enables someone to upload aws specific master keys which needs to be pushed to a different uri compared to what we use by default 

Also update the error to include the error code response ========================================================================================================================================================================

## by making the following changes

< DESCRIBE YOUR CHANGES >

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [x] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

**Review by mkesavan13 — APPROVED** [review]
Looks good to me. Thanks for triaging & fixing this one @arun3528

**Review by torgeadelin — APPROVED** [review]
would be good to also test the KmsError changes, but looking good otherwise

File: packages/@webex/internal-plugin-encryption/src/kms-errors.js [hunk 1]
```diff
@@ -50,6 +50,14 @@ export class KmsError extends Exception {
       message += `\nKMS_REQUEST_ID: ${body.requestId}`;
     }
 
+    if (body.statusCode) {
+      message += `\nKMS_STATUS_CODE: ${body.statusCode}`;
+    }
+
+    if (body.errorCode) {
+      message += `\nKMS_ErrorCode: ${body.errorCode}`;
+    }
```
Comment (InteractiveTimmy) [inline packages/@webex/internal-plugin-encryption/src/kms-errors.js:59]: potentially needs a test for these conditional branches. Nonblocking.

**Review by InteractiveTimmy — APPROVED** [review]
LGTM, not too familiar with this section of KMS though.

File: packages/@webex/internal-plugin-encryption/src/kms-errors.js (hunk 1 above)
Comment (arun3528) [reply packages/@webex/internal-plugin-encryption/src/kms-errors.js:59]: fixed it

**Pull request merged**